[zacwong2151] iP

zacwong2151 opened the pull request:
# DukePro

> "This is great. I love software engineering." – David Allen [source](https://www.google.com/imgres?imgurl=https%3A%2F%2Fbmj2k.files.wordpress.com%2F2016%2F12%2Frandom.png&tbnid=Me3I1nKkOAGynM&vet=12ahUKEwiEtanmqpuBAxVkmmMGHTC0D7gQMygCegQIARBv..i&imgrefurl=https%3A%2F%2Fbmj2k.com%2F2016%2F12%2F04%2Fa-random-google-search%2F&docid=IZhe8ghL8t6E_M&w=564&h=400&q=random%20image%20on%20google&ved=2ahUKEwiEtanmqpuBAxVkmmMGHTC0D7gQMygCegQIARBv)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. Download it from [here](https://github.com/zacwong2151/ip/releases/tag/tag-A-Jar).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

And it is **FREE**!

## Features:

- [x] Managing tasks
- [x] Managing deadlines 
- [ ] Reminders (*coming soon*)

#

If you're a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

zacwong2151 committed:
added couple print.ln statements for level-0

zacwong2151 committed:
finished Level-1

zacwong2151 committed:
added add, list features for level-2

zacwong2151 committed:
added Task class and added Mark as Done feature

zacwong2151 committed:
added multiple subclasses of Task, made Task abstract

zacwong2151 committed:
added automated Text UI Testing

zacwong2151 committed:
implemented handle exceptions

zacwong2151 committed:
implemented delete feature for Level-6

zacwong2151 committed:
Level-7

zacwong2151 committed:
Level-8

zacwong2151 committed:
doing this commit to merge Level-8

zacwong2151 committed:
deleted duke.txt file to resolve conflict

zacwong2151 committed:
halfway doing A-MoreOOP

zacwong2151 committed:
A-MoreOOP

zacwong2151 committed:
Added untracked files

zacwong2151 committed:
Merge branch 'A-MoreOOP'

zacwong2151 committed:
added A-Packages

zacwong2151 committed:
Merge branch 'A-Packages'

zacwong2151 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

zacwong2151 committed:
A-Gradle

zacwong2151 committed:
Merge branch 'A-Gradle'

zacwong2151 committed:
A-Jar

zacwong2151 committed:
Merge branch 'A-Jar'

zacwong2151 committed:
A-JUnit

zacwong2151 committed:
Merge branch 'A-JUnit'

zacwong2151 committed:
A-JavaDoc

zacwong2151 committed:
commit changes to data/temp.txt

zacwong2151 committed:
resolve conflict

zacwong2151 committed:
duke.txt stuff

zacwong2151 committed:
A-CodingStandard

zacwong2151 committed:
Merge branch 'A-CodingStandard'

zacwong2151 committed:
testing branching

zacwong2151 committed:
is this the subject body?

zacwong2151 committed:
Merge branch 'branch-Level-9'

Carlintyj commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,344 @@
+     * @param string the input string
+     * @throws DukeException if input is invalid
+     */
+    public void markDescription(String string) throws DukeException {
```
Perhaps a better name would be "userInput" instead of "string"

Carlintyj commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,344 @@
+     * @param string of the duke.Task
+     * @return the description of the duke.Task
+     */
+    public static String getDescription(String string) {
```
Likewise here as well, maybe there can be a better parameter naming instead of "string"

Carlintyj commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,344 @@
+     *
+     * @return The number of tasks in the collection
+     */
+    public int getSize() {
```
Would it be better if instead of getSize() you use getTaskCount() instead?

Carlintyj commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,344 @@
+     * @param i The position of the duke.Task
+     * @return The duke.Task
+     */
+    public Task getTask(int i) {
```
Would this be better if it was named getTaskAtIndex(int index) ?

Carlintyj reviewed (commented):
LGTM, just some nits to fix.

zacwong2151 committed:
added resources and image folder

zacwong2151 committed:
learnt JavaFX basics, next gonna create GUI for chatbot

zacwong2151 committed:
created GUI for chatbot

zacwong2151 committed:
updated .gitignore file

VN-Hao commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,45 @@
+        return this.description;
+    }
+
+    public void markAsUnDone() {
```
I think you should rename the method to "markAsUndone".

VN-Hao replied:
I think you should rename this method to `markAsUndone`.

nubnubyas commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,89 @@
+package duke;
+import java.io.File;
```
Perhaps you can have a line of space between the package and the import? Personally, I think it would have better readability. The same goes for all the other files.

VN-Hao commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+
+    /**
+     * constructor to initialise a Deadline object
+     * @param description the Task description that is obtained from the Task class
+     * @param by the deadline time component that is stored as a LocalDateTime
```
I think you should add punctuation behind each parameter description as stated [here](https://se-education.org/guides/conventions/java/intermediate.html#comments). Moreover, there should also be an empty line between description and parameter section.

You can apply these rules for the rest of your JavaDoc!

nubnubyas reviewed (commented):
LGTM.
Really good job for IP. I really like how you have JavaDoc for all the functions and even between lines of code to make it easy to understand your code. 
Just a personal opinion, I would prefer to have different packages, for example:
ToDo, Deadline, Event, and Task can be under the same package together. I think this makes the code more organised.
Overall, the code quality is great and I really like it.

VN-Hao reviewed (commented):
LGTM! 🥳

VN-Hao commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+public class Deadline extends Task {
+    private LocalDateTime by;
+
+    /**
+     * constructor to initialise a Deadline object
+     * @param description the Task description that is obtained from the Task class
+     * @param by the deadline time component that is stored as a LocalDateTime
+     */
```
I think you should add punctuation behind each parameter description as stated [here](https://se-education.org/guides/conventions/java/intermediate.html#comments). Moreover, there should also be an empty line between description and parameter section.

You can apply these rules for the rest of your JavaDoc!

VN-Hao reviewed (commented):
Overall, LGTM! However, there are still some violations in method namings and comment conventions.😄

zacwong2151 committed:
improved OOP in code

zacwong2151 committed:
Merge branch 'Level-10'

zacwong2151 committed:
added basic GUI CSS

zacwong2151 committed:
Examine code and refactor to improve code quality

After implementing the GUI chatbot function, there is a lot of code
left over from the text-based chatbot.

This needs to be removed as there is a lot of unnecessary code that is
cluttering up the code base

In order to make my code more readable and human-friendly, let's remove
those unnecessary code and also employ good coding standard practices

zacwong2151 committed:
Merge pull request #1 from zacwong2151/A-CodeQuality

Examine code and refactor to improve code quality

zacwong2151 committed:
Add Undo and Help feature

Users are unable to undo a command that they ran previously. Users are
also unable to see example of correct user inputs, and require in-App
guidance

This needs to be changed so as to improve the user experience!

Let's
1. Implement a undo command
2. Implement a help comman

note*: wrap each line at maximum 72 characters

zacwong2151 committed:
Merge branch 'Undo-feature'

zacwong2151 committed:
unimportant changes

zacwong2151 committed:
Add tag for A-CodeQuality

Let's add a commit with unimportant changes to the code so that I can
push a A-CodeQuality tag to my remote repo

zacwong2151 committed:
Merge pull request #3 from zacwong2151/A-CodeQuality

Add tag for A-CodeQuality

zacwong2151 committed:
Merge branch 'master' of https://github.com/zacwong2151/ip

zacwong2151 committed:
Add some assertions

zacwong2151 committed:
Encapsulate Ui for various classes

zacwong2151 committed:
Merge branch 'master' into A-Assertions

zacwong2151 committed:
Add assertions to TaskList

zacwong2151 committed:
Merge pull request #4 from zacwong2151/A-Assertions

Merge A-Assertions

zacwong2151 committed:
Add more JUnit testing

zacwong2151 committed:
Update README.md

zacwong2151 committed:
Update README.md

zacwong2151 committed:
Add finishing touches

I cannot execute the jar file due to the error: JavaFX runtime
components missing

Let's finally fix it by working around classpath issues by changing how
the main method in Main.java is called. Let's also touch up our code and
bring it to its final version.

zacwong2151 committed:
Fix conflicts in README

zacwong2151 committed:
Fix pathing issues

zacwong2151 committed:
Change paths to use java Paths

zacwong2151 committed:
My last commit (hopefully) :)

The code does not handle the case where the data files do not exist at
the start. When someone else takes my chatbot and runs it for the first
time, the required data files will not exist in their computer

Let's change this by checking if those files exist, and create them if
they do not. This is done whenever the application is opened. Thus
users will always have the required data files to save their tasks and
commands to memory

zacwong2151 committed:
Add Ui.png file

zacwong2151 committed:
Add dependencies
